src/main/kotlin/domain/result/GameProfit.kt

```diff
@@ -0,0 +1,34 @@
+package domain.result
+
+import domain.money.Money
+import domain.money.Profit
+import domain.person.Participants
+import domain.person.Player
+
+class GameProfit(
+    private val bettingMoneys: Map<Player, Money>,
+) {
+
+    fun getPlayersProfit(participants: Participants) =
+        participants.players.associateWith { player ->
+            val playerBettingMoney = requireNotNull(bettingMoneys[player]) { BETTING_MONEY_NOT_FOUND_ERROR }
+            player.getPlayerProfit(participants.dealer.state, playerBettingMoney)
+        }
+
+    fun getDealersProfit(participants: Participants): Profit {
+        val result = participants.players.sumOf { player ->
+            val playerBettingMoney = requireNotNull(bettingMoneys[player]) { BETTING_MONEY_NOT_FOUND_ERROR }
+            participants.dealer.getPlayerProfit(player.state, playerBettingMoney).value
```

Comment:
```suggestion
            participants.dealer.getPlayerProfit(player.state, playerBettingMoney).value
```
GameProfit 클래스는 participants, dealer, profit등 너무 많은 것을 알고 있어요.
디미터 법칙이 무엇인지 학습해보시고, 어떻게 개선하면 좋을지 고민해보면 어떨까요?